py3status/modules/screenshot.py

```diff
@@ -4,82 +4,179 @@
-                self.upload_path):
-            command = 'scp %s/%s%s %s@%s:%s' % (
-                self.save_path, file_name, '.jpg', self.upload_user,
-                self.upload_server, self.upload_path)
-            subprocess.Popen(command.split())
+        button = event['button']
+        if button == 1:
+            self._full_text = "Shooting"
+            self._color = self.py3.COLOR_PROCESSING or self.py3.COLOR_DEGRADED
+            if self.random_name:
+                file_name = "%s.%s" % (self._filename_generator(self.file_length),
+                                       self.image_format)
+            else:
+                timestamp = datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+                file_name = "%s.%s" % (timestamp, self.image_format)
+
+            command = '%s %s/%s' % (self.screenshot_command,
+                                    self.save_path,
+                                    file_name)
+
+            subprocess.call(command.split())
+            logstring = None
+
+            if self.push:
+                self._full_text = "Sending"
+                if self.post_processing == "scp":
+                    logstring = self._push_scp(file_name)
+                elif self.post_processing == "lutim":
+                    logstring = self._push_lutim(file_name)
+                elif self.post_processing is not None:
+                    post_process = self.post_processing.split(sep=':', maxsplit=1)
+                    if post_process[0] == "custom" and len(post_process) == 2:
+                        logstring = self._push_custom(file_name, post_process[1])
+            logline = "%s -> %s\n" % (file_name, logstring) if logstring else file_name
+
+            if self.logfile:
+                f = open(self.logfile, 'a')
+                f.write(logline)
+                f.close()
+
+            self._full_text = file_name
+            self._color = self.py3.COLOR_DONE or self.py3.COLOR_GOOD
+            self.py3.prevent_refresh()
+        else:
+            self._reset_text()
+
+    def _reset_text(self):
+        self._full_text = self.empty_text
+        self._color = None
+
+    def _push_scp(self, file_name):
+        if self.upload_server and self.upload_user and self.upload_path:
+            dest = '%s@%s:%s' % (self.upload_user, self.upload_server, self.upload_path)
+            command = 'scp %s/%s %s' % (self.save_path, file_name, dest)
+            subprocess.call(command.split())
+            logstring = "SCP to %s" % dest
+            return logstring
+
+    def _push_lutim(self, file_name):
+        if self.upload_server:
+            import requests
+            r = requests.post(self.upload_server,
+                              data={'format': 'json', 'delete-day': 1},
+                              files={'file': open("%s/%s" % (self.save_path,
+                                                             file_name), 'rb')},
+                              )
```

Comment:
I'm happy with a general something went wrong rather than every problem being caught individually.  Hopefully we will get #459 finished at some point and it will basically be a nicely wrapped 'requests' type thing and do the error issues.